vernac/declare.ml

```diff
@@ -1919,6 +1940,10 @@ end = struct
             Termops.Internal.debug_print_constr (EConstr.of_constr t) ++ str ".")
 
   let declare_mutdef ~uctx ~pinfo pe i CInfo.{ name; impargs; typ; _} =
+    let pe =
+      if i > 0 && not pe.proof_entry_opaque then
+        Internal.map_entry_body pe ~f:(fun (x,_) -> x,Evd.empty_side_effects)
+      else pe in
```

Comment:
You mean `prepare_proof`? I think I had to do it here because in the case of mutual stuff we have only one proof entry which is declared many times (I was surprised by that). The "multiplicity" is dictated by the `pinfo.Proof_info.cinfo` see the last line of declare_mutdef.

Reply:
Yes, that's a big problem which is not new to this PR, maybe we should move the comment you remove in the place this was taken care before and add it here?

Because this sees like a fixme to me.

Reply:
What do you want me to write? The command was saying that one should not duplicate side effects and that maybe the right thing to do is to declare them once with the first constant. This is what the patch does, no? Bringing it back seems odd, since that issue is solved.

Cleaning up to the code so that we generate one proof entry per copy of the mutfix is OK to me, but maybe in another PR?

Reply:
I don't know what to write as I don't understand the current code generating this now.